praw/models/reddit/live.py

```diff
@@ -281,6 +281,55 @@ def close(self):
         url = API_PATH['live_close'].format(id=self.thread.id)
         self.thread._reddit.post(url)
 
+    def update(self, **settings):
+        """Update settings of the live thread.
+
+        The following settings are supported:
+
+        :param title: (Optional) The title of the live thread.
+        :param description: (Optional) The live thread's description.
+        :param nsfw: (Optional) Indicate whether this thread is not safe for
+            work.
+        :param resources: (Optional) Markdown formatted information that is
+            useful for the live thread.
+        :returns: If any settings are given, return new :class:`.LiveThread`
+            instance (updated settings can be retrieved through the instance.)
+            If no settings are given, raise ``TypeError``.
+
+        Unspecified settings will maintain their current value.
+
+        Usage:
+
+        .. code-block:: python
+
+           thread = reddit.live('xyu8kmjvfrww')
+           new_title = '[nsfw]' + thread.title
+           old_description = thread.description
+           updated_thread = thread.contrib.update(title=new_title, nsfw=True)
+           assert updated_thread.title == new_title
+           assert updated_thread.description == old_description
+
+        """
+        available_settings = ('title', 'description', 'nsfw', 'resources')
+        settings = {key: settings[key] for key in available_settings
+                    if key in settings}
+
+        if not settings:
+            raise TypeError('No settings are given.')
+        elif len(settings) == len(available_settings):
```

Comment:
While I like this approach to minimize a potential request, if we want to do this I'd prefer using two separate methods as was done with `Subreddit`, one for `update` which can fill in existing values, and one for `set` which directly sets all known attributes with the option to pass in additional.

The primary reason is that I'd like to avoid PRAW needing to be updated when Reddit adds additional setting fields.